### PR TITLE
Chore: run pytest in CI

### PR DESCRIPTION
**Pull request by afeld**
Also add configuration to run Django tests in GitHub Actions. Want a baseline test that works as a starting place for adding tests in https://github.com/cal-itp/benefits/pull/445. Merged in https://github.com/cal-itp/benefits/pull/455 to build off of those.

### CODE REVIEW

File: .github/workflows/tests-django.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      - name: Check out code
+        uses: actions/checkout@v3
+
+      - name: Create environment file
+        run: cp .env.sample .env
+        working-directory: .devcontainer
+
+      - name: Build base image
+        run: docker compose build client
+        working-directory: .devcontainer
```
Comment (afeld) [inline .github/workflows/tests-django.yml:18]: I wanted to keep this as simple as possible (to start) so didn't bother with the fancier Buildx-based Docker config that other workflows have; we can switch to that now or later.

**Comment by afeld** [comment]
Getting the same test failure here as locally:

```
  File "/home/calitp/app/benefits/core/views.py", line 36, in index
    buttons[0].classes.append("mt-3")
IndexError: list index out of range
```

This seems to mean that no agencies are active…?

https://github.com/cal-itp/benefits/blob/1f8b89f933d75c6d4b4e8a240a61fcd4b3ec17bc/benefits/core/views.py/#L34-L37

Assuming this is because the database isn't set up or something?

**Comment by thekaveman** [comment]
We started a branch to add these types of tests [`django-tests-spike`](https://github.com/cal-itp/benefits/tree/django-test-spike) - can you take a look and build on that set up?

**Comment by afeld** [comment]
Current error:

```
$ python manage.py migrate
…
sqlite3.OperationalError: unable to open database file
```

My theory is this is a volume issue, where the volume isn't writable from the Docker engine that the Compose container is running in.

**Comment by thekaveman** [comment]
Putting this in draft while it is still in progress - so we don't get notified for review every change 😁

**Comment by thekaveman** [comment]
My strong advice would be - get the tests to pass locally first, we can worry about CI after. As you're running up against, it can be more of a pain to get the Docker stuff all working together in CI.

But if they are passing locally at least we know we have them and they "should" work.

**Comment by afeld** [comment]
> so we don't get notified for review every change

Sorry about the noise! Have been unable to reproduce the error locally, hence needing to do so many pushes.

**Comment by thekaveman** [comment]
I'm thinking more... does the container/app actually need to be running to run these tests?

Would it work to have job steps that install requirements, run `bin/init.sh`, and run pytest -- directly in the GH Actions runner?

**Comment by afeld** [comment]
> run pytest -- directly in the GH Actions runner

Good call. I was keep consistent by sticking with the container-based setup, but this was much easier.

File: .github/workflows/tests-pytest.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+name: Pytest
+
+on: push
+
+jobs:
+  run:
```
Comment (thekaveman) [inline .github/workflows/tests-pytest.yml:6]: Sorry, one change - can we make this `pytest` or something - easier to reference in required checks and other places.

**Review by thekaveman — CHANGES_REQUESTED** [review]

File: .github/workflows/tests-pytest.yml (hunk 2 above)
Comment (afeld) [reply .github/workflows/tests-pytest.yml]: Oh, strange it doesn't include the workflow name. Done!

**Review by thekaveman — APPROVED** [review]
Thank you, this is super helpful!

After merge I'll update the branch protection to require this check as well.

**Pull request merged**